packages/rrweb/src/replay/index.ts

```diff
@@ -402,9 +426,37 @@ export class Replayer {
         );
       }, 1);
     }
-    if (this.service.state.context.events.find(indicatesTouchDevice)) {
-      this.mouse.classList.add('touch-device');
-    }
+    this.service.state.context.events.forEach((e: eventWithTime) => {
+      if (indicatesTouchDevice(e)) {
+        const pointerId = this.getPointerId(e.data);
+        if (!this.pointers[pointerId]) {
+          this.createPointer(pointerId);
+        }
+
+        const pointer = this.pointers[pointerId];
+        pointer.pointerEl.classList.add('touch-device');
+      }
+    });
+  }
+
+  private getPointerId(
+    d: incrementalData | mousemoveData | mouseInteractionData,
+  ): number {
+    const pointerId =
+      'pointerId' in d && typeof d.pointerId === 'number' ? d.pointerId : -1;
+    return pointerId;
+  }
```

Comment:
nit: we could pull this out of the class since it doesn't depend on any internal class instance state (e.g. `this`)